### PR TITLE
feat(bestax-migrate): react-bulma-components → bestax-bulma codemod CLI, skill, and docs

### LINKED ISSUE #332: [Feature] bestax-migrate: codemod CLI + agent skill to migrate apps off react-bulma-components
## Affected Package(s)

- [x] Other: new workspace package `bestax-migrate` (npm: `bestax-migrate`)
- [x] bulma-ui (`@allxsmith/bestax-bulma`) — one packaging fix found along the way
- [x] docs (`@allxsmith/bestax-docs`)
- [x] Other: `skills/` (new `bestax-migrate` skill)

## Description

`react-bulma-components` (couds) is unmaintained — last release 2022, stuck on Bulma 0.9.x — and its users have no assisted path onto an actively maintained Bulma v1 library. Give them one:

- **`bestax-migrate`** — a jscodeshift-based codemod CLI (`pnpm dlx bestax-migrate react-bulma-components src/`) that rewrites imports, components, props, responsive objects, **stylesheets** (CSS imports → `bestax.css`; SCSS `@import 'bulma/bulma'` + `$var !default` → `@use 'bulma/sass' with (…)` + extras), and **package.json** (RBC out, bestax-bulma in, bulma → ^1, node-sass → dart sass). Anything unsafe gets an inline `// TODO(bestax-migrate)` comment plus an end-of-run report — never a silent skip or a guessed rewrite.
- **Multi-source architecture** (`src/sources/registry.ts`) so future codemods for other libraries slot in as new source folders.
- **`skills/bestax-migrate`** — an agent skill that drives the codemod and resolves every TODO from mapping references. Deliberately *not* bundled into create-bestax (existing sites only).
- **Docs** — a migration guide under Getting Started → Migration plus the skills listing.

Constraints: react-bulma-components is never installed anywhere in the repo (supply-chain policy) — fixtures and validation corpora are text-only; migrated *output* is typechecked against the workspace bestax-bulma. Validation includes real-world apps (openaedmap-frontend, python-events, i8008-tools) migrated to a green production build with before/after screenshot comparison.

### PR DESCRIPTION
**Pull request by allxsmith**
# Pull Request

## Description

Adds **`bestax-migrate`**, a new workspace package publishing the `bestax-migrate` CLI: a jscodeshift codemod that migrates existing apps from the unmaintained `react-bulma-components` (v4, Bulma 0.9) to `@allxsmith/bestax-bulma` (Bulma v1). Multi-source by design (`src/sources/registry.ts`) so future library codemods slot in as new source folders.

- [x] bulma-ui (`@allxsmith/bestax-bulma`) — packaging fix (see below)
- [ ] create-bestax (`create-bestax`)
- [x] docs (`@allxsmith/bestax-docs`)
- [x] Other (please specify): new `bestax-migrate` workspace package; `skills/bestax-migrate`; repo wiring (pnpm-workspace, commitlint scope, turbo, CI release step)

**What the codemod converts** — imports (named/namespace/`const { Input } = Form` destructuring), all 32 RBC v4 components and compounds via a data-driven mapping table, prop renames/value conversions, responsive breakpoint objects, structural rewrites (Navbar dropdowns, Card.Image, Field grouping, icon-font children → `<Icon name…>`), **stylesheets** (CSS imports → `bestax.css`; SCSS `@import 'bulma/bulma'` + `$var !default` overrides → `@use 'bulma/sass' with (…)` + extras, including Parcel-style relative node_modules paths), and **package.json** (RBC removed, bestax-bulma added, bulma → ^1.0.4, node-sass → dart sass ^1.79, `peer-deps` warnings for React <18 / Font Awesome <6). Anything unsafe gets an inline `// TODO(bestax-migrate)` comment + a report entry — never a silent skip. Files in unparseable formats (`.astro`/`.vue`/`.svelte`/`.mdx`) importing RBC are reported as `unsupported-file`.

**Also in this PR** — `fix(bulma-ui)`: removed the `typesVersions` wildcard that remapped the root `types` field onto itself, which broke type resolution for every `moduleResolution: node10` consumer (surfaced by migrating a real app).

**Supply-chain**: react-bulma-components is **never installed** anywhere in this repo. Fixtures and validation corpora are text-only; the kitchen-sink e2e typechecks migrated *output* against the workspace bestax-bulma. The corpus script is deliberately not in CI.

## Related Issue(s)

Closes #332

## Type of Change

- [x] New feature
- [x] Bug fix (bulma-ui `typesVersions`)
- [x] Documentation
- [x] Build tooling

## Checklist

- [x] My code follows the project style guidelines
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works (184 unit/e2e tests, 13 fixture pairs, kitchen-sink e2e with `tsc --noEmit` against built bulma-ui; coverage over the package thresholds)
- [x] I have added/updated documentation as needed (migration guide, skills listing, `bestax-migrate/CLAUDE.md`, README)
- [ ] I have updated Storybook stories as needed (bulma-ui) — n/a, no component changes
- [x] My changes require a change to the documentation
- [x] All new and existing tests passed (`pnpm all` equivalent gates: build, typecheck, test+coverage, lint, format, conformance)
- [x] Any relevant dependencies are updated (`@babel/parser` added — declared, already in the lockfile transitively)
- [x] If this PR changes commands, conventions, or package structure, the affected `CLAUDE.md` files are updated (root + new `bestax-migrate/CLAUDE.md`)

## Screenshots / Demos

Validated against real frontend-only RBC v4 apps, all reaching a green production build post-migration:

| App | Stack | Result |
|---|---|---|
| openaedmap-frontend | Vite 7, TS, React 19 | ✓ builds, **zero manual edits** |
| python-events | Vite, TS, React 19, FA 5 | ✓ builds (`--legacy-peer-deps` for their FA 5 pin — now predicted by the `peer-deps` report) |
| i8008-tools emulator UI | Parcel, JS, React 17, SCSS | ✓ builds (React 17→18 upgrade — now predicted; sass ≥1.79 pin) |

Before/after screenshot comparison: i8008 (SCSS path) is pixel-identical; python-events (CSS path) differs only in the documented `bestax.css` primary color (`#1e6b99`) and Bulma v1 spacing. Story-level visual comparison of the RBC storybook corpus (49 stories/side) was reviewed during development.

## Additional Context

- Independent semantic-release: `bestax-migrate` scope, tag `bestax-migrate@x.y.z`; first `feat` on main publishes 1.0.0. The `fix(bulma-ui)` commit patches `@allxsmith/bestax-bulma` independently.
- `@allxsmith/bestax-bulma` is a runtime dependency of bestax-migrate (`workspace:^`) so the published package registers as an npm dependent and dlx runs pull the library through the registry.
- The skill ships via `npx skills add https://github.com/allxsmith/bestax --skill bestax-migrate` and is intentionally **not** synced into create-bestax.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_017yX1c4wgFY9knAAVzWNPJu

---
_Generated by [Claude Code](https://claude.ai/code/session_017yX1c4wgFY9knAAVzWNPJu)_

## Summary by CodeRabbit

* **New Features**
  * Added the `bestax-migrate` CLI to migrate React Bulma apps from `react-bulma-components` to Bestax Bulma on Bulma v1, including component, prop, responsive, stylesheet, and dependency updates.
  * Added dry-run previews with end-of-run reports, TODO markers for manual work, and checks that flag unsupported inputs.

* **Documentation**
  * Published migration guides and a dedicated migration skill with reference tables covering components, props, CSS, and remaining TODOs.
  * Updated repo and skills documentation to include `bestax-migrate`.

* **Tests/Quality**
  * Added CLI/transform/unit and end-to-end “kitchen-sink” coverage, plus mapping-coverage validation.

* **Chores**
  * Improved CI coverage publishing and release automation for `bestax-migrate`, along with repo formatting/lint exclusions.

### CODE REVIEW

File: bestax-migrate/src/sources/react-bulma-components/specials.ts [hunk 1]
```diff
@@ -0,0 +1,799 @@
...
+        ),
+        'Heading'
+      );
+      const replacement = plainElement(
+        ctx.j,
+        'p',
+        'heading',
+        rest,
+        element.children ?? []
+      );
```
[the remark this answers is not in the document]
Comment (allxsmith) [reply bestax-migrate/src/sources/react-bulma-components/specials.ts:242]: Fixed in 499eecb. Confirmed against the code — all six sites dropped a user `className` while `plain-loader` merged it.

The fix extracts plain-loader's treatment into a shared `mergeClassName(ctx, path, element, base, where)` helper and applies it everywhere a plain element is emitted:

- **heading** → `<p className="heading extra">` (string literal merged)
- **plain-label / plain-help** → merged into the computed `label is-*` / `help is-*` class
- **panel-tab** → merged with `is-active` when present, or emitted alone (`<a className="custom">`)
- **breadcrumb-item** → merged into the `<li>` class (where RBC puts the item's className)
- **table-container** (multi-child) → `<div className="table-container scroller">`
- **table-container** (single-child fold): there's no wrapper left to carry the class, so a `prop:className` TODO is emitted instead of a silent drop
- A dynamic `className` at any site keeps the base class and leaves a `prop:className` TODO (same as plain-loader did)

Also audited the remaining `!== 'className'` filter sites: `card-image` keeps className on the Card.Image wrapper and `tab-item` keeps it on the Tabs.Item element — no drops there.

Fixtures regenerated via the built transform (not hand-edited); `edge-cases-2` now exercises the literal-merge, merge-without-base, dynamic-TODO, and single-child-fold cases. All 185 tests, coverage, lint, typecheck, format, and the real-world corpus validation pass.

---
_Generated by [Claude Code](https://claude.ai/code)_
